File: utils/python/aoc/input.py
import sys
from os import path

class Input:
  """Puzzle input."""

  def __init__(self, raw: str) -> None:
    self.__raw = raw

  @property
  def raw(self) -> str:
    """Raw input."""
    return self.__raw

  @property
  def lines(self) -> list[str]:
    """Lines of the input. Strips whitespace."""
    return self.__raw.strip().splitlines()

  @property
  def ints(self) -> list[int]:
    """Ints found in the input, one int per line."""
    return [int(x) for x in self.lines]

input = Input("")

directory = path.dirname(path.abspath(sys.modules["__main__"].__file__ or ""))
with open(path.join(directory, "input.txt"), "r") as file:
  input = Input(file.read())

File: utils/deno/input.ts
import * as path from "https://deno.land/std/path/mod.ts";

const directory = path.dirname(path.fromFileUrl(Deno.mainModule));

const raw = await Deno.readTextFile(path.resolve(directory, "input.txt"))

/** Input represents a puzzle's input. */
export class Input {
  constructor(public readonly raw: string) {}

  /** The lines of the input. Trims trailing whitespace. */
  get lines() {
    return this.raw.trimEnd().split("\n")
  }

  /** The numbers of the input, one number per line. */
  get numbers() {
    return this.lines.map(x => Number(x))
  }
}

/** The puzzle's input */
const input = new Input(raw)
export default input
